lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -236,10 +236,11 @@ public void suspendAll(ErrorInfo error, boolean notifyStateChange) {
 
         /**
          * By spec RTN15c3
+         * Move queued messages from connection manager to their respective channel
          * @param queuedMessages Queued messages transferred from connection
          */
         @Override
-        public void reattachOnResumeFailure(List<ConnectionManager.QueuedMessage> queuedMessages) {
+        public void transferQueuedMessages(List<ConnectionManager.QueuedMessage> queuedMessages) {
```

Comment:
As written, this is very inefficient. We're iterating through all queued messages multiple times, once for each channel.

Reply:
aa1bc86296f22e21386630394f75f81ae22675c0